pkg/util/stringutil/stringutil_test.go

```diff
@@ -0,0 +1,144 @@
+package stringutil
+
+import (
+	"fmt"
+	"math/rand"
+	"runtime"
+	"runtime/debug"
+	"sync"
+	"testing"
+)
+
+var oldBank sync.Map
+
+// This is the old implementation of the string bank to use for comparison benchmarks
+func BankLegacy(s string) string {
+	ss, _ := oldBank.LoadOrStore(s, s)
+	return ss.(string)
+}
+
+func ClearBankLegacy() {
+	oldBank = sync.Map{}
+}
+
+func copyString(s string) string {
+	return string([]byte(s))
+}
+
+func generateBenchData(totalStrings, totalUnique int) []string {
+	randStrings := make([]string, 0, totalStrings)
+
+	// create totalUnique unique strings
+	for i := 0; i < totalUnique; i++ {
+		randStrings = append(randStrings, fmt.Sprintf("%s/%s/%s", RandSeq(10), RandSeq(10), RandSeq(10)))
+	}
+
+	// set the seed such that the resulting "remainder" strings are deterministic for each bench
+	rand.Seed(1523942)
+
+	// append a random selection from 0-totalUnique to the list.
+	for i := 0; i < totalStrings-totalUnique; i++ {
+		randStrings = append(randStrings, copyString(randStrings[rand.Intn(totalUnique)]))
+	}
+
+	// shuffle the list of strings
+	rand.Shuffle(totalStrings, func(i, j int) { randStrings[i], randStrings[j] = randStrings[j], randStrings[i] })
+
+	return randStrings
+}
+
+func benchmarkLegacyStringBank(b *testing.B, totalStrings, totalUnique int) {
+	b.StopTimer()
+	randStrings := generateBenchData(totalStrings, totalUnique)
+
+	for i := 0; i < b.N; i++ {
+		b.StartTimer()
```

Comment:
For a future PR, I think this is more idiomatic (unless I'm missing something!):
```suggestion
	randStrings := generateBenchData(totalStrings, totalUnique)
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
```

Reply:
Does Reset() call Start() on an already Stopped() bench?

Reply:
My _assumption_ is that a new call to a `BenchmarkFoo` func (if being handled by the bench runner for a higher-`b.N` bench) should have a totally fresh timer. But... can't say for sure.